common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1874,6 +1874,13 @@ def _update_item_from_fields(
                 block_data.fields = settings
 
                 new_id = new_structure['_id']
+
+                # When a draft block from a published structure is reverted in revert_to_published(),
+                # the update_version is updated while source_version is not. If source_version is not None,
+                # set it to None - so that has_changes() will use update_version for comparison and return True.
+                # Prevents the block from getting stuck in the published state.
+                block_data.edit_info.source_version = None
```

Comment:
"`source_version` records which revision a block was copied from. In this method, we're updating the block, so it's no longer a direct copy, and we can remove the `source_version` reference."